comparenweso.py

```diff
@@ -0,0 +1,191 @@
+            iter_list = iter_list[:-1]
+
+        return verify_wesolowski(
+            str(discriminant),
+            str(x[0]),
+            str(x[1]),
+            str(y[0]),
+            str(y[1]),
+            str(proof[-1][0]),
+            str(proof[-1][1]),
+            iterations,
+        )
+    except Exception:
+        return False
+
+def oldie(
+    disc,
+    a,
+    b,
+    proof,
+    iter,
+    sb,
+    witness,
+    doold
+):
+    if z == 0:
+        x = ClassGroup.from_ab_discriminant(int(a), int(b), int(disc))
+    
+        return check_proof_of_time_nwesolowski(
+            int(disc),
+            x,
+            proof,
+            iter,
+            sb,
+            witness,
+        )
+        
+    return verify_n_wesolowski(
+        disc,
+        a,b,
+        proof,
+        iter,
+        sb,
+        witness,
+    )
+
+for z in range(2):
+    iters = 1000
+    t1 = time.time()
+    for i in range(iters):
+        is_valid = oldie(
+            str(-131653324254138636653163861414331698305531090221496467927360326686715180966094250598321899621249972220387687148397451395672779897144571112116763666653213748473909547482437246405018707472153290116227072825447643324530509016778432769802300913461285128339119844239772697652504835780459732685000796733645621728639),
+            str(2),
+            str(1),
+            bytes.fromhex("003f360be667de706fe886f766fe20240de04fe2c2f91207f1bbdddf20c554ab8d168b2ce9664d75f4613375a0ab12bf8158983574c9f5cd61c6b8a905fd3fa6bbffc5401b4ccedbe093b560293263a226e46302e720726586251116bc689ef09dc70d99e0a090c4409f928e218e85032fdbee02fedd563073be555b75a70a2d6a430033bc7a4926e3504e87698a0ace0dee6364cced2e9142b4e4cbe55a6371aab41e501ceed21d79d3a0dbbd82ce913c5de40b13eb7c59b1b52b6ef270ee603bd5e7fffcc9f5fae6dbd5aeec394181af130c0fdd195b22be745449b7a584ac80fc75ed49acfdb4d650f5cd344f86377ebbbaef5b19a0af3ae08101d1697f5656a52193000000000071c6f40024c342868a0c2a201b1b26a5d52c5d2f92a106c19ff926deb3fba1e74a444ecee3f8f507c062b949a2eaadd442b049417f82e8811526fa83c6d099d75323e068ffeca9dcd163761000c65d21dede72787ac350f25bdd3d29db6e9cb0e22c8124c724db33660c88784e2871b62ecf816846db7b469c71cad9a5dcfc5548ed2dd781006fa15b968facf4d79219646267eb187a670306d1ff1a59fc28ae00d36bb5a1cba659f48aa64a9022711a66105ef14401ff3948add265240aaad329ee76ba4c2300496746b86bcccacff5947c3fcb956cde2cffae10435960d7097f989aac742cf1047887f11584d20297958385e1715fe0f9b69141750c20d8134420eafec68fd10000000001555540006958aabfe4cc5d870e61fef82bcf1f2c3859e2bd8b1177e8a8872376b5cabace5dcb59b6fecada7e522d05f6f0e352939a6bfdf8c454fbe822cfa5ce97d17be0ffde44a4812cde9d04ec5c08dce6f9146586fdc8e081e05ec690b7effe24ea756f3d300f361203b61e1a39220c6eafa7852842674e317dcae5549c78c7144296ff004a6d0d2854c55e4c1de2f17dc4480b81652cfec37124ef41560a28c853482732434d1c006763b2e341528ae0bcc29fb76f1a4dafd99ade4fd75ec9cc9ca3f3d7001bcb6eb71e43eb22169ab721637551a8ec93838eb0825e9ecba9175297a00b146e9fdd244c5b722f29d3c46ec38840ba18f1f06ddec3dea844867386c2e1ac95"),
```

Comment:
do you have a reference to where these test vectors come from?

Reply:
this is from dev chain from the logs